### PR TITLE
enh 434: instance template should support multiple NICs

### PR DESCRIPTION
**Pull request by alexkonkin**

### CODE REVIEW

**Comment by alexkonkin** [comment]
> 
> 
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here (e.g. `I signed it!`) and we'll verify it.
> #### What to do if you already signed the CLA
> ##### Individual signers
> 
>     * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> 
> ##### Corporate signers
> 
>     * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google.com/docs/cla/#troubleshoot)).
> 
>     * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
>     * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2FGoogleCloudPlatform%2Fcloud-foundation-toolkit%2Fpull%2F31) for more info**.

I signed it!

File: dm/templates/instance/examples/instance.yaml [hunk 1]
```diff
@@ -14,7 +14,8 @@ resources:
       diskImage: projects/ubuntu-os-cloud/global/images/family/ubuntu-1804-lts
       machineType: f1-micro
       diskType: pd-ssd
-      network: default
```
Comment (ocsig) [inline dm/templates/instance/examples/instance.yaml]: Please update your implementation in a way that both 'network' (as the current implementation) as 'networks' (the new implementation) is supported to assure backwards compatibility.

File: dm/templates/instance/instance.py.schema [hunk 2]
```diff
@@ -25,53 +25,61 @@ required:
   - zone
   - machineType
   - diskImage
-  - network
```
Comment (ocsig) [inline dm/templates/instance/instance.py.schema]: Please update your implementation in a way that both 'network' (as the current implementation) as 'networks' (the new implementation) is supported to assure backwards compatibility.

File: dm/templates/instance_template/instance_template.py.schema [hunk 3]
```diff
@@ -20,7 +20,7 @@ info:
 
 required:
   - diskImage
-  - network
```
Comment (ocsig) [inline dm/templates/instance_template/instance_template.py.schema]: Please update your implementation in a way that both 'network' (as the current implementation) as 'networks' (the new implementation) is supported to assure backwards compatibility.

File: dm/templates/instance/tests/integration/instance_1_nic.yaml [hunk 4]
```diff
@@ -17,9 +17,20 @@ resources:
       machineType: f1-micro
       diskType: pd-ssd
       canIpForward: true
-      network: test-network-${RAND}
-      subnetwork: regions/us-central1/subnetworks/test-subnet-${RAND}
+      networks:
```
Comment (akopachevskyy-globallogic) [inline dm/templates/instance/tests/integration/instance_1_nic.yaml]: why we don't use "network" in this case?

Comment (alexkonkin) [reply dm/templates/instance/tests/integration/instance_1_nic.yaml]: Andriy, this test is dedicated for the networks property.
We have a separate test that should also check the network property.

**Comment by ocsig** [comment]
Please update the Managed Instance Group Schema to support the multi NIC Instance templates

**Pull request merged**